presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -473,6 +606,35 @@ private static void checkLongitude(double longitude, String errorMessage)
         checkCondition(longitude >= MIN_LONGITUDE && longitude <= MAX_LONGITUDE, errorMessage);
     }
 
+    private static double correctLatitude(double latitude)
+    {
+        if (latitude > MAX_LATITUDE)
+            return MAX_LATITUDE;
+
+        if (latitude < MIN_LATITUDE)
+            return MIN_LATITUDE;
+
+        return latitude;
+    }
+
+    private static double correctLongitude(double longitude)
+    {
+        if (longitude > MAX_LONGITUDE)
+            return MAX_LONGITUDE;
+
+        if (longitude < MIN_LONGITUDE)
+            return MIN_LONGITUDE;
+
+        return longitude;
+    }
+
+    private static boolean pointWithinCircle(double radiusInKm, Point center, Point toBeChecked) {
+        if ( GeoFunctions.greatCircleDistance(center.getY(), center.getX(), toBeChecked.getY(), toBeChecked.getX()) <= radiusInKm )
```

Comment:
Calling this function incurs extra costs of checking input parameters and computing `sin(center.latitude)`, `cos(center.latitude)` and `toRadians(center.longitude)` repeatedly for each tile being checked. Is there a way to eliminate or reduce unnecessary computation?

Reply:
Im not sure this is unnecessary computation based on the requirement. I did add breaks as soon as we found a tile that was included in the column to avoid unnecessary checks.

The other alternative we had discussed was to not exclude at all and just return all the tiles between top and bottom and left and right. Im not sure if thats still something we can consider which would remove all the exclude tile logic. Depends on how precise the tiles returned need to be?

Reply:
@avinashidnani `GeoFunctions.greatCircleDistance` function starts with the following checks:

```
        checkLatitude(latitude1);
        checkLongitude(longitude1);
        checkLatitude(latitude2);
        checkLongitude(longitude2);
```
These checks are not necessary.

Also, the function applies the following computation to `lattitude1` on each invocation. In this case `lattitude1` is the same for all tiles/calls (it is center.latitude). 

```
        double radianLatitude1 = toRadians(latitude1);

        double sin1 = sin(radianLatitude1);
        double cos1 = cos(radianLatitude1);
```